### PR TITLE
🌐 Add Japanese translation for `docs/tutorial/security/oauth2-jwt.md`

### PR DESCRIPTION
**Pull request by sattosan**
Relates to https://github.com/tiangolo/fastapi/issues/1572

This translates tutorial/security/oauth2-jwtr.md into Japanese.

### CODE REVIEW

**Review by tokusumi — CHANGES_REQUESTED** [review]
Thank you for your great work.
I suggested a few proposals to improve them.

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+これでセキュリティの流れが全てわかったので、<abbr title="JSON Web Tokens">JWT</abbr>トークンと安全なパスワードのハッシュ化を使用して、実際にアプリケーションを安全にしてみましょう。
+
+このコードは、アプリケーションで実際に使用したり、パスワードハッシュをデータベースに保存するといった用途に利用できます。
+
+本章では、前章の続きから始めて、コードをアップデートしていきます。
+
+## JWT について
+
+JWTとは"JSON Web Token"の略称です。
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md]: thanks!!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
+```
+
+これらは暗号化されていないので、誰でもコンテンツから情報を復元できてしまいます。
+
+しかし、トークンは署名されているため、あなたが発行したトークンを受け取った人は、あなたが実際に発行したということを検証できます。
+
+例えば、1週間の有効期限を持つトークンを作成したとします。ユーザーが翌日そのトークンを持って戻ってきたとき、そのユーザーはまだシステムにログインしていることがわかります。
+
+1週間後、トークンが期限切れとなるとどうなるでしょうか？ユーザーは認証されず、新しいトークンを得るために再びサインインしなければなりません。また、ユーザー（または第三者）がトークンを修正して有効期限を変更しようとした場合、署名が一致しないため、トークンの修正を検知できます。
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:25]: I see, that is a mistranslation, thanks!!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+ここでは、推奨されているものを使用します：<a href="https://cryptography.io/" class="external-link" target="_blank">pyca/cryptography</a>。
+
+!!! tip "豆知識"
+    このチュートリアルでは以前、<a href="https://pyjwt.readthedocs.io/" class="external-link" target="_blank">PyJWT</a>を使用していました。
+
+    しかし、Python-joseは、PyJWTのすべての機能に加えて、他のツールと統合して構築する際に必要となるいくつかの追加機能を提供しています。そのため、代わりにPython-joseを使用するように更新されました。
+
+## パスワードのハッシュ化
+
+"ハッシュ化"とは、あるコンテンツ（ここではパスワード）を、規則性のないバイト列（単なる文字列）に変換することです。
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:54]: thanks!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+例えば、「車」や「ブログ記事」を識別することができます。
+
+そして、「ドライブ」（車の場合）や「編集」（ブログの場合）など、そのエンティティに関する権限も追加できます。
+
+また、JWTトークンをユーザー（またはボット）に渡すことができます。ユーザーは、JWTトークンを使用するだけで、アカウントを持っていなくても、APIが生成したJWTトークンを使ってそれらの行動（車の運転、ブログ投稿の編集）を実行することができるのです。
+
+これらのアイデアを使用すると、JWTをより高度なシナリオに使用できます。
+
+しかしながら、それらのエンティティのいくつかが同じIDを持つ可能性があります。例えば、 `foo`（ユーザー`foo`、車 `foo`、ブログ投稿` foo`）などです。
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:190]: thanks!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+開発者ツールを開くと、送信されるデータにはトークンだけが含まれており、パスワードはユーザーを認証してアクセストークンを取得する最初のリクエストでのみ送信され、その後は送信されないことがわかります。
+
+<img src="/img/tutorial/security/image10.png">
+
+!!! note "備考"
+    ヘッダーの`Authorization`には、`Bearer`で始まる値があります。
+
+## `scopes` を使った高度なユースケース
+
+OAuth2には、"スコープ"という概念があります。
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:238]: thanks!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+
+サーバーを実行し、ドキュメントに移動します：<a href="http://127.0.0.1:8000/docs" class="external-link" target="_blank">http://127.0.0.1:8000/docs</a>
+
+次のようなユーザーインターフェイスが表示されます：
+
+<img src="/img/tutorial/security/image07.png">
+
+前回と同じ方法でアプリケーションの認証を行います。
+
+認証情報の使用：
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:206]: thanks!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+!!! note "備考"
+    ヘッダーの`Authorization`には、`Bearer`で始まる値があります。
+
+## `scopes` を使った高度なユースケース
+
+OAuth2には、"スコープ"という概念があります。
+
+これらを利用して、JWTトークンに特定の権限セットを追加することができます。
+
+そして、このトークンをユーザーに直接、または第三者に与えて、制限付きでAPIを操作することができます。
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:242]: I see, thanks!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+
+例えば、1週間の有効期限を持つトークンを作成したとします。ユーザーが翌日そのトークンを持って戻ってきたとき、そのユーザーはまだシステムにログインしていることがわかります。
+
+1週間後、トークンが期限切れとなるとどうなるでしょうか？ユーザーは認証されず、新しいトークンを得るために再びサインインしなければなりません。また、ユーザー（または第三者）がトークンを修正して有効期限を変更しようとした場合、署名が一致しないため、トークンの修正を検知できます。
+
+JWT トークンを使って遊んでみたいという方は、こちらをチェック <a href="https://jwt.io/" class="external-link" target="_blank">https://jwt.io</a>。
+
+## `python-jose` のインストール
+
+PythonでJWTトークンを生成して検証するには、`python-jose`をインストールする必要があります：
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:31]: The meaning of "or" rather than "and" seems to be closer. thanks!

**Comment by sattosan** [comment]
@tokusumi 
Thanks for pointing out all the details!
All fixed!!😀

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+!!! tip "豆知識"
+    このチュートリアルでは以前、<a href="https://pyjwt.readthedocs.io/" class="external-link" target="_blank">PyJWT</a>を使用していました。
+
+    しかし、Python-joseは、PyJWTのすべての機能に加えて、他のツールと統合して構築する際に必要となるいくつかの追加機能を提供しています。そのため、代わりにPython-joseを使用するように更新されました。
+
+## パスワードのハッシュ化
+
+"ハッシュ化"とは、あるコンテンツ（ここではパスワード）を、規則性のないバイト列（単なる文字列）に変換することです。
+
+特徴として、全く同じ内容（全く同じパスワード）を渡しても、全く同じような規則性のないバイト列が得られます。
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:56]: thanks!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 10]
```diff
@@ -0,0 +1,266 @@
...
+
+しかし、規則性のないバイト列からパスワードに戻すことはできません。
+
+### パスワードのハッシュ化を使う理由
+
+データベースが盗まれても、ユーザーの平文のパスワードは盗まれず、ハッシュ値だけが盗まれます。
+
+そして、泥棒はそのパスワードを別のシステムで使おうとしますができません（多くのユーザーはどこでも同じパスワードを使用しているため、危険性があります）。
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:64]: thanks!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+
+## `passlib` のインストール
+
+PassLib は、パスワードのハッシュを処理するための優れたPythonパッケージです。
+
+このパッケージは、多くの安全なハッシュアルゴリズムとユーティリティをサポートします。
+
+推奨されるアルゴリズムは"Bcrypt"です。
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:72]: I see, thanks!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 12]
```diff
@@ -0,0 +1,266 @@
...
+
+使用するかどうかは任意ですが、`sub`にユーザーの識別情報を入れることになるので、ここでは使用しています。
+
+JWTは、ユーザーを識別して、そのユーザーがAPI上で直接操作を実行できるようにする以外にも、他の用途で使用されることがあります。
+
+例えば、「車」や「ブログ記事」を識別することができます。
+
+そして、「ドライブ」（車の場合）や「編集」（ブログの場合）など、そのエンティティに関する権限も追加できます。
+
+また、JWTトークンをユーザー（またはボット）に渡すことができます。ユーザーは、JWTトークンを使用するだけで、アカウントを持っていなくても、APIが生成したJWTトークンを使ってそれらの行動（車の運転、ブログ投稿の編集）を実行することができるのです。
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:186]: Thanks!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 13]
```diff
@@ -0,0 +1,266 @@
...
+
+
+## パスワードのハッシュ化と検証
+
+必要なツールを `passlib`からインポートします。
+
+PassLib の「context」を作成します。これは、パスワードのハッシュと検証に使用されるものです。
+
+!!! tip "豆知識"
+    PassLibのcontext には、非推奨の古いものだけを含む様々なハッシュアルゴリズムを使用した検証機能もあります。
```
Comment (sh0nk) [inline docs/ja/docs/tutorial/security/oauth2-jwt.md:101]: Need to add translation of "only to allow verifying them (passwords), etc."
Actually this implies "it should not be used for hashing any more", but the original sentence does not have it, so skipping.

```suggestion
    PassLibのcontextには、検証だけが許された非推奨の古いハッシュアルゴリズムを含む、様々なハッシュアルゴリズムを使用した検証機能もあります。
```

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 14]
```diff
@@ -0,0 +1,266 @@
...
+    例えば、Djangoアプリケーションからデータベース内の同じデータをFastAPIアプリケーションと共有できるだけではなく、同じデータベースを使用してDjangoアプリケーションを徐々に移行することもできます。
+
+    また、ユーザーはDjangoアプリまたは**FastAPI**アプリからも、同時にログインできるようになります。
+
+
+## パスワードのハッシュ化と検証
+
+必要なツールを `passlib`からインポートします。
+
+PassLib の「context」を作成します。これは、パスワードのハッシュと検証に使用されるものです。
```
Comment (sh0nk) [inline docs/ja/docs/tutorial/security/oauth2-jwt.md:98]:
```suggestion
PassLib の「context」を作成します。これは、パスワードのハッシュ化と検証に使用されるものです。
```

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 15]
```diff
@@ -0,0 +1,266 @@
...
+
+受け取ったトークンを復号して検証し、現在のユーザーを返します。
+
+トークンが無効な場合は、すぐにHTTPエラーを返します。
+
+```Python hl_lines="89-106"
+{!../../../docs_src/security/tutorial004.py!}
+```
+
+## `/token` パス操作の更新
```
Comment (sh0nk) [inline docs/ja/docs/tutorial/security/oauth2-jwt.md:164]: In FastAPI docs, "path operation" is a special term translated into "パスオペレーション": e.g. https://fastapi.tiangolo.com/tutorial/first-steps/#step-3-create-a-path-operation

```suggestion
## `/token` パスオペレーションの更新
```

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 16]
```diff
@@ -0,0 +1,266 @@
...
+
+## 確認
+
+サーバーを実行し、ドキュメントに移動します：<a href="http://127.0.0.1:8000/docs" class="external-link" target="_blank">http://127.0.0.1:8000/docs</a>
+
+次のようなユーザーインターフェイスが表示されます：
+
+<img src="/img/tutorial/security/image07.png">
+
+前回と同じ方法でアプリケーションの認証を行います。
```
Comment (sh0nk) [inline docs/ja/docs/tutorial/security/oauth2-jwt.md:204]:
```suggestion
前回と同じ方法でアプリケーションの認可を行います。
```

**Review by sh0nk — COMMENTED** [review]
Thank you for your great effort for this big document!

**Comment by sh0nk** [comment]
Also I'd appreciate if you can join reviewing my works as you commented there @sattosan ;)
https://github.com/tiangolo/fastapi/pull/2631, https://github.com/tiangolo/fastapi/pull/2681

**Review by tokusumi — APPROVED** [review]
approved for these changes!
I'd love to that would be polished by sh0nk's review.

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 17]
```diff
@@ -0,0 +1,266 @@
...
+</div>
+
+また、<a href="https://github.com/mpdavis/python-jose" class="external-link" target="_blank">Python-jose</a>だけではなく、暗号を扱うためのパッケージを追加で必要とします。
+
+ここでは、推奨されているものを使用します：<a href="https://cryptography.io/" class="external-link" target="_blank">pyca/cryptography</a>。
+
+!!! tip "豆知識"
+    このチュートリアルでは以前、<a href="https://pyjwt.readthedocs.io/" class="external-link" target="_blank">PyJWT</a>を使用していました。
+
+    しかし、Python-joseは、PyJWTのすべての機能に加えて、他のツールと統合して構築する際に必要となるいくつかの追加機能を提供しています。そのため、代わりにPython-joseを使用するように更新されました。
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:50]: thanks!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md [hunk 18]
```diff
@@ -0,0 +1,266 @@
...
+
+## パスワードのハッシュ化
+
+「ハッシュ化」とは、あるコンテンツ（ここではパスワード）を、規則性のないバイト列（単なる文字列）に変換することです。
+
+特徴として、全く同じ内容（全く同じパスワード）を渡すと、全く同じ規則性のないバイト列に変換されます。
+
+しかし、規則性のないバイト列からパスワードに戻すことはできません。
```
[the remark this answers is not in the document]
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:58]: thanks!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md (hunk 13 above)
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:101]: thanks!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md (hunk 14 above)
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:98]: thanks!

File: docs/ja/docs/tutorial/security/oauth2-jwt.md (hunk 16 above)
Comment (sattosan) [reply docs/ja/docs/tutorial/security/oauth2-jwt.md:204]: thanks!

**Comment by sattosan** [comment]
@sh0nk 
Thanks for pointing out all the details!
All fixed!!😀

**Review by sh0nk — APPROVED** [review]
Verified that all are fixed! LGTM!

**Comment by tiangolo** [comment]
Awesome, thanks a lot @sattosan! :bow: 

And thank you for the reviews @tokusumi and @sh0nk! :rocket: :coffee:

**Pull request merged**